web3/main.py

```diff
@@ -152,13 +153,53 @@ def currentProvider(self):
         ))
         return self.manager.providers[0]
 
-    @coerce_return_to_text
-    def sha3(self, value, encoding="hex"):
-        if encoding == 'hex':
-            hex_string = value
+    @staticmethod
+    @apply_to_return_value(encode_hex)
+    def sha3(primitive=None, text=None, hexstr=None, encoding=None):
```

Comment:
I think we should either:

* implement validation that ensures that you can't call this function *wrongly* such as `web3.sha3(b'arst', text='bar', hexstr='baz')`
* do some shenanigans with the function signature to make it impossible to call in that manner.

Reply:
I'm anti-shenanigan (at least in this context), will validate.